traits/tests/test_rich_compare.py

```diff
@@ -186,3 +189,20 @@ def test_assumptions(self):
         self.assertEqual(self.a.name, self.same_as_a.name)
         self.assertNotEqual(self.a.name, self.different_from_a.name)
         return
+
+
+class OldRichCompareTestCase(unittest.TestCase):
+    def test_raises_runtime_warning(self):
+        with warnings.catch_warnings(record=True) as warn_msgs:
+            warnings.simplefilter("always", RuntimeWarning)
+
+            class OldRichCompare(HasTraits):
+                bar = Any(rich_compare=False)
+
+        self.assertEqual(len(warn_msgs), 1)
+        warn_msg = warn_msgs[0]
+        self.assertIn(
+            "'rich_compare' kwarg has been removed",
+            str(warn_msg.message)
+        )
+        self.assertIn("trait_handlers", warn_msg.filename)
```

Comment:
I don't think this is desirable: it's not useful for the user if the warning refers to the trait internals. Instead, the filename should point to the code that *uses* `rich_compare`, in this case this test file (`test_rich_compare.py`), so `"trait_handlers"` should be replaced with `"test_rich_compare"` and the `stacklevel` argument to the warning should be adjusted to whatever's necessary to make the test pass.

Reply:
I've fixed this to check that the name of the module (so in this case "test_rich_compare") appears in the filename, and adjusted the stacklevel at the point where the `DeprecationWarning` is issue accordingly.